### PR TITLE
Update colab examples

### PR DESCRIPTION
**Pull request by wenxindongwork**
# Use transformers' AutoModelForCausalLM instead of optimum-tpu's  AutoModelForCausalLM for finetuning. 

The `from optimum.tpu` version imports models that are specifically optimized for inference. While the colab example works for smaller models, it fails with a HBM OOM error for llama3-70b (on a v4-256). Changing the following import statement solved the problem.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests?

### CODE REVIEW

**Review by allenwang28 — APPROVED** [review]

**Review by tengomucho — APPROVED** [review]
You are right @wenxindongwork, at some point I added some improvements for tuning, but with FSDP the models are essentially the same in `transformers` and in `optimum.tpu`, so it might well be easier just to import the `transformers` version.
Note that on the other hand, this means that the model might be loaded in `float32` and end up using more memory, whereas the `optimum.tpu`'s models will load on `bfloat16` by default. If you do this, you can end up with OOM on some configurations.

File: examples/language-modeling/llama_tuning.md [hunk 1]
```diff
@@ -69,7 +68,11 @@ data = data.map(lambda samples: tokenizer(samples["quote"]), batched=True)
 You then need to specify the FSDP training arguments to enable the sharding feature,the function will deduce the classes that should be sharded:
 
 ```python
-fsdp_training_args = fsdp_v2.get_fsdp_training_args(model)
+cls_to_wrap = "LlamaDecoderLayer"
+fsdp_training_args = {
+    "fsdp": "full_shard",
+    "fsdp_config": fsdp_v2.get_fsdp_config(cls_to_wrap),
+}
```
Comment (tengomucho) [inline examples/language-modeling/llama_tuning.md:75]: same comment here, I think it's simpler just to call `get_fsdp_training_args` for these supported models.

**Review by tengomucho — CHANGES_REQUESTED** [review]

File: examples/language-modeling/gemma_tuning.ipynb [hunk 2]
```diff
@@ -297,7 +297,11 @@
     "from transformers import TrainingArguments\n",
     "\n",
     "# Set up the FSDP arguments\n",
-    "fsdp_training_args = fsdp_v2.get_fsdp_training_args(model)\n",
+    "cls_to_wrap = \"GemmaDecoderLayer\"\n",
+    "fsdp_training_args = {\n",
+    "    \"fsdp\": \"full_shard\",\n",
+    "    \"fsdp_config\": fsdp_v2.get_fsdp_config(cls_to_wrap),\n",
+    "}\n",
```
[the remark this answers is not in the document]
Comment (wenxindongwork) [reply examples/language-modeling/gemma_tuning.ipynb]: thanks for the quick review! `get_fsdp_training_args` accepts only the optimum.tpu model class, not the transformers one. I updated the `get_fsdp_training_args function`, so it should now work.

**Review by tengomucho — COMMENTED** [review]
Nice, with your change it looks fine. 🤗
Last question I was wondering was about using the `AutoModelForCausalLM` class from transformers rather than from optimum.tpu. 
AS I mentioned in a comment, in Optimum TPU models we set dtype to `bfloat16`, because TPU are capable of that and it uses less memory, e.g. [here](https://github.com/huggingface/optimum-tpu/blob/426d7be437ca367327114bc10f9badf1a0d1d056/optimum/tpu/modeling_llama.py#L1655). 
I hadn't test the scripts with float32, but if you do and it works fine please let me know, so we can just merge this.

**Comment by wenxindongwork** [comment]
Just updated the examples to load the models in bf16 instead, hope that works!

**Review by tengomucho — APPROVED** [review]

**Comment by tengomucho** [comment]
The code style workflow is failing, can you run it locally (`make style`) and push again so we can merge this please?

**Comment by wenxindongwork** [comment]
just installed ruff and ran make sytle. Thanks!

**Review by tengomucho — APPROVED** [review]

**Pull request merged**